src/main/kotlin/com/criptext/mail/scenes/mailbox/data/GetPendingLinkRequestWorker.kt

```diff
@@ -0,0 +1,101 @@
+package com.criptext.mail.scenes.mailbox.data
+
+import com.criptext.mail.api.HttpClient
+import com.criptext.mail.api.models.Event
+import com.criptext.mail.api.models.UntrustedDeviceInfo
+import com.criptext.mail.bgworker.BackgroundWorker
+import com.criptext.mail.bgworker.ProgressReporter
+import com.criptext.mail.db.models.ActiveAccount
+import com.github.kittinunf.result.Result
+import com.github.kittinunf.result.flatMap
+import org.json.JSONArray
+import java.io.IOException
+
+class GetPendingLinkRequestWorker(
+        httpClient: HttpClient,
+        activeAccount: ActiveAccount,
+        override val publishFn: (
+                MailboxResult.GetPendingLinkRequest) -> Unit)
+    : BackgroundWorker<MailboxResult.GetPendingLinkRequest> {
+
+    private val apiClient = MailboxAPIClient(httpClient, activeAccount.jwt)
+
+    override val canBeParallelized = false
+
+    override fun catchException(ex: Exception): MailboxResult.GetPendingLinkRequest =
+             MailboxResult.GetPendingLinkRequest.Failure()
+
+    override fun work(reporter: ProgressReporter<MailboxResult.GetPendingLinkRequest>)
+            : MailboxResult.GetPendingLinkRequest? {
+        val result = fetchPendingEvents()
+                .flatMap(parseEvents)
+                .flatMap(processEvents)
+        return when (result) {
+            is Result.Success -> {
+                MailboxResult.GetPendingLinkRequest.Success(result.value)
+            }
+            is Result.Failure -> {
+                catchException(result.error)
+            }
+        }
+    }
+
+    private fun fetchPendingEvents():Result<String, Exception> {
+        return Result.of {
+            val responseText = apiClient.getPendingEvents()
+            if (responseText.isEmpty()) "[]" else responseText
+        }
+    }
+
+    private val parseEvents: (String) -> Result<List<Event>, Exception> = { jsonString ->
```

Comment:
This is already done elsewhere. Why not create a `EventLoader` class that can reuse that logic?